File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System.Web
Imports System.Web.UI
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpress.Web

Namespace UploadControl.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function
		Public Function UploadControlCallbackAction() As ActionResult
            UploadControlExtension.GetUploadedFiles("uc", UploadControlDemosHelper.ValidationSettings, AddressOf UploadControlDemosHelper.uc_FileUploadComplete)
			Return Nothing
		End Function
	End Class

	Public Class UploadControlDemosHelper
		Public Const UploadDirectory As String = "~/Content/UploadControl/UploadFolder/"

		Public Shared ReadOnly ValidationSettings As New UploadControlValidationSettings() With {.AllowedFileExtensions = New String() {".jpg", ".jpeg", ".jpe", ".gif", ".bmp"}, .MaxFileSize = 20971520}

		Public Shared Sub uc_FileUploadComplete(ByVal sender As Object, ByVal e As FileUploadCompleteEventArgs)
			If e.UploadedFile.IsValid Then
				Dim resultFilePath As String = HttpContext.Current.Request.MapPath(UploadDirectory & e.UploadedFile.FileName)
				'e.UploadedFile.SaveAs(resultFilePath, True)'Code Central Mode - Uncomment This Line
				Dim urlResolver As IUrlResolutionService = TryCast(sender, IUrlResolutionService)
				If urlResolver IsNot Nothing Then
					e.CallbackData = urlResolver.ResolveClientUrl(resultFilePath)
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web;
using System.Web.UI;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using DevExpress.Web.ASPxUploadControl;

namespace UploadControl.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        public ActionResult UploadControlCallbackAction() {
            UploadControlExtension.GetUploadedFiles("uc", UploadControlDemosHelper.ValidationSettings, UploadControlDemosHelper.uc_FileUploadComplete);
            return null;
        }
    }

    public class UploadControlDemosHelper {
        public const string UploadDirectory = "~/Content/UploadControl/UploadFolder/";

        public static readonly ValidationSettings ValidationSettings = new ValidationSettings {
            AllowedFileExtensions = new string[] { ".jpg", ".jpeg", ".jpe", ".gif", ".bmp", },
            MaxFileSize = 20971520,
        };

        public static void uc_FileUploadComplete(object sender, FileUploadCompleteEventArgs e) {
            if(e.UploadedFile.IsValid) {
                string resultFilePath = HttpContext.Current.Request.MapPath(UploadDirectory + e.UploadedFile.FileName);
                //e.UploadedFile.SaveAs(resultFilePath, true);//Code Central Mode - Uncomment This Line
                IUrlResolutionService urlResolver = sender as IUrlResolutionService;
                if(urlResolver != null) {
                    e.CallbackData = urlResolver.ResolveClientUrl(resultFilePath);
                }
            }
        }
    }
}